Replaced AppError with StudioAppError in Error page and removed vueti…

Abhishek-Punhani opened the pull request:
<!--
 1. Following guidance below, replace …'s with your own words
 2. After saving the PR, tick of completed checklist items
 3. Skip checklist items that are not applicable or not necessary
 4. Delete instruction/comment blocks
-->

## Summary
<!--
 * description of the change
 * manual verification steps performed
 * screenshots if the PR affects the UI
-->
- Replaced `AppError` with `StudioAppError` (created in issue #5235 )
-  Replaced `ActionLink` with `KRouterLink` using appearance="raised-button" and primary props

<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/9ee37e9e-3442-42cb-b622-565622684981" />

## References
<!--
 * references to related issues and PRs
 * links to mockups or specs for new features
 * links to the diffs for any dependency updates, e.g. in iceqube or the perseus plugin
-->
#5295 

## Reviewer guidance
<!--
 * how can a reviewer test these changes?
 * are there any risky areas that deserve extra testing
-->
- In ChannelListIndex, temporarily update the condition that displays the error to:
``` 
  <ChannelListAppError
    v-if="true"
    :error="{ errorType: 'UNAUTHORIZED' }"
  />
```
- Login as user@a.com with password a
- Go to Channels

MisRob commented:
Hi @Abhishek-Punhani, thank you. Code changes make sense and the button works as expected.

Before I merge, could you update the button label as requested in the issue?

> Replace ActionLink by primary KRouterLink with raised button appearance and 'Back to home' label

Abhishek-Punhani commented:
> Hi @Abhishek-Punhani, thank you. Code changes make sense and the button works as expected.
> 
> Before I merge, could you update the button label as requested in the issue?
> 
> > Replace ActionLink by primary KRouterLink with raised button appearance and 'Back to home' label

Yes @MisRob , I have removed the text in slot and added it as a `:text` prop.

MisRob commented:
>  I have removed the text in slot and added it as a :text prop.

That's fine @Abhishek-Punhani. My comment was about content of label: It needs to say _'Back to home'_ :)

MisRob commented:
No worries. It's  bit of extra task in the issue to align it with other pages.

Abhishek-Punhani commented:
@MisRob , I had made the suggested changes. Please let me know if it's fine now!
